File: xyz/CodeForces/1B/src/Solution.cpp
#include <cctype>
#include <iostream>
#include <stack>
#include <string>

using namespace std;

void process(string coordinate) {
    int column = 0;
    int row = 0;

    int length = coordinate.size();

    int k = 1;
    if (coordinate[0] == 'R' && !isalpha(coordinate[k])) {
        while (k < length && isdigit(coordinate[k])) {
            k += 1;
        }
    }

    if (k == 1 || k == length) {
        k = 0;
        while (isalpha(coordinate[k])) {
            column = column * 26 + (static_cast<int>(coordinate[k]) - 64);
            k += 1;
        }
        while (k < length) {
            row = row * 10 + (static_cast<int>(coordinate[k]) - 48);
            k += 1;
        }

        cout << "R" << row << "C" << column << endl;
    } else {
        k = 1;
        while (isdigit(coordinate[k])) {
            row = row * 10 + (static_cast<int>(coordinate[k]) - 48);
            k += 1;
        }
        k += 1;
        while (k < length) {
            column = column * 10 + (static_cast<int>(coordinate[k]) - 48);
            k += 1;
        }

        stack<char> s;

        int r;
        while (column != 0) {
            r = column % 26;
            column = column / 26;
            if (r == 0) {
                s.push('Z');
                column -= 1;
            } else {
                s.push(static_cast<char>(r + 64));
            }
        }

        while (!s.empty()) {
            cout << s.top();
            s.pop();
        }
        cout << row << endl;
    }
}

int main() {
    int testCases;

    string coordinate;

    while (cin >> testCases) {
        for (int i = 0; i < testCases; i++) {
            cin >> coordinate;
            process(coordinate);
        }
    }
}

File: xyz/CodeForces/1B/src/Solution.java
import java.util.Scanner;
import java.util.Stack;

public class Solution {

    public static void process(String coordinate) {
        int column = 0;
        int row = 0;

        int length = coordinate.length();

        int k = 1;
        if (coordinate.charAt(0) == 'R' && !Character.isLetter(coordinate.charAt(k))) {
            while (k < length && Character.isDigit(coordinate.charAt(k))) {
                k++;
            }
        }

        if (k == 1 || k == length) {
            k = 0;
            while (Character.isLetter(coordinate.charAt(k))) {
                column = column * 26 + ((int) coordinate.charAt(k) - 64);
                k++;
            }
            while (k < length) {
                row = row * 10 + ((int) coordinate.charAt(k) - 48);
                k++;
            }

            System.out.printf("R%dC%d\n", row, column);
        } else {
            k = 1;
            while (Character.isDigit(coordinate.charAt(k))) {
                row = row * 10 + ((int) coordinate.charAt(k) - 48);
                k++;
            }
            k++;
            while (k < length) {
                column = column * 10 + ((int) coordinate.charAt(k) - 48);
                k++;
            }

            Stack<Character> s = new Stack<>();

            int r;
            while (column != 0) {
                r = column % 26;
                column = column / 26;
                if (r == 0) {
                    s.push('Z');
                    column--;
                } else {
                    s.push((char) (r + 64));
                }
            }

            while (!s.empty()) {
                System.out.print(s.peek());
                s.pop();
            }
            System.out.println(row);
        }
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int testCases;

        String coordinate;

        while (input.hasNext()) {
            testCases = input.nextInt();
            for (int i = 0; i < testCases; i++) {
                coordinate = input.next();
                process(coordinate);
            }
        }
    }
}
